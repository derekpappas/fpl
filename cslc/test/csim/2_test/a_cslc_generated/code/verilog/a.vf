//THIS CODE WAS GENERATED USING THE FASTPATHLOGIC HDL COMPILER
//COPYRIGHT (c) 2006, 2007 FastpathLogic Inc
//OUTPUT FILE NAME  : a.vf
//FILE GENERATED ON : Fri Nov 13 03:38:37 2009

a.v
stim_expect_mem_template.v
tb_testbench.v
