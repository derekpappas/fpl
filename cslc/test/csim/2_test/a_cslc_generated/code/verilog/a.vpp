`include "defines.v"

module a(clk,
         reset_,
         in,
         valid_in,
         out,
         valid_out
         );

// Location of source csl unit: file name = 2.csl line number = 1
  input clk;
  input reset_;
  input  [3 - 1:0] in;
  input            valid_in;
  output [3 - 1:0] out;
  output           valid_out;
  reg    [3 - 1:0] out;
  reg              valid_out;
  `include "a.logic.vh"
endmodule

