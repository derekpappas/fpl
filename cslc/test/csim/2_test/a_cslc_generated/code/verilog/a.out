#! /usr/local/bin/vvp
:ivl_version "0.9.1" "(v0_9_1)";
:vpi_time_precision + 0;
:vpi_module "system";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_0xb67def0 .scope module, "tb" "tb" 2 3;
 .timescale 0 0;
P_0xb663a68 .param/l "EXP_ADDR_WIDTH" 2 16, +C4<010000>;
P_0xb663a90 .param/l "EXP_MEM_WIDTH" 2 15, +C4<01>;
P_0xb663ab8 .param/str "EXP_VECTOR_FILE" 2 21, "exp";
P_0xb663ae0 .param/l "EXP_VECTOR_ID" 2 17, +C4<0>;
P_0xb663b08 .param/l "EXP_VECTOR_MAX_ERR" 2 23, +C4<0>;
P_0xb663b30 .param/str "EXP_VECTOR_NAME" 2 19, "exp";
P_0xb663b58 .param/l "EXP_VECTOR_RADIX" 2 22, +C4<0>;
P_0xb663b80 .param/l "EXP_VECTOR_VERSION" 2 18, +C4<0>;
P_0xb663ba8 .param/l "SIM_TIMEOUT_CNT" 2 5, +C4<01100100>;
P_0xb663bd0 .param/l "STIM_ADDR_WIDTH" 2 7, +C4<010000>;
P_0xb663bf8 .param/l "STIM_MEM_WIDTH" 2 6, +C4<01>;
P_0xb663c20 .param/str "STIM_VECTOR_FILE" 2 12, "stim";
P_0xb663c48 .param/l "STIM_VECTOR_ID" 2 8, +C4<0>;
P_0xb663c70 .param/l "STIM_VECTOR_MAX_ERR" 2 14, +C4<0>;
P_0xb663c98 .param/str "STIM_VECTOR_NAME" 2 10, "stim";
P_0xb663cc0 .param/l "STIM_VECTOR_RADIX" 2 13, +C4<0>;
P_0xb663ce8 .param/l "STIM_VECTOR_VERSION" 2 9, +C4<0>;
L_0xb6aab40 .functor XOR 1, C4<z>, L_0xb6abe90, C4<0>, C4<0>;
L_0xb6aac40 .functor XNOR 1, C4<z>, L_0xb6abe90, C4<0>, C4<0>;
v0xb6a9c50_0 .net/s *"_s0", 31 0, C4<00000000000000000000000001100100>; 1 drivers
v0xb6a9d10_0 .net "a0_in_in", 0 0, L_0xb6ab1f0; 1 drivers
v0xb6a9de0_0 .net "a0_out_out", 0 0, C4<z>; 0 drivers
v0xb6a9e60_0 .net "a0_out_out_expect", 0 0, L_0xb6abe90; 1 drivers
v0xb6a9f10_0 .net "a0_out_out_match_en", 0 0, L_0xb6aac40; 1 drivers
v0xb6a9f90_0 .net "a0_out_out_mismatch_en", 0 0, L_0xb6aab40; 1 drivers
v0xb6aa050_0 .var "clk", 0 0;
v0xb6aa0d0_0 .var/i "cycle_cnt", 31 0;
v0xb6aa1a0_0 .var "exp_a0_match_count", 15 0;
v0xb6aa240_0 .var "exp_a0_mismatch_count", 15 0;
v0xb6aa340_0 .var "exp_a0_transaction_count", 15 0;
RS_0x2aaaab4e9468 .resolv tri, L_0xb6abab0, L_0xb6ac7c0, C4<z>, C4<z>;
v0xb6aa3e0_0 .net8 "expect_out_valid", 0 0, RS_0x2aaaab4e9468; 2 drivers
v0xb6aa4d0_0 .var/i "file_mcd", 31 0;
RS_0x2aaaab4e9258 .resolv tri, L_0xb6abb10, L_0xb6ac820, C4<z>, C4<z>;
v0xb6aa550_0 .net8 "id_err", 0 0, RS_0x2aaaab4e9258; 2 drivers
v0xb6aa650_0 .var "rand_valid", 0 0;
v0xb6aa6d0_0 .net "rd_en", 0 0, v0xb6aa650_0; 1 drivers
v0xb6aa7e0_0 .net "stop_sim", 0 0, L_0xb6aaa50; 1 drivers
v0xb6aa860_0 .var "testbench_reset", 0 0;
RS_0x2aaaab4e9528 .resolv tri, L_0xb6ab910, L_0xb6ac620, C4<z>, C4<z>;
v0xb6aa980_0 .net8 "version_err", 0 0, RS_0x2aaaab4e9528; 2 drivers
L_0xb6aaa50 .cmp/ge.s 32, v0xb6aa0d0_0, C4<00000000000000000000000001100100>;
S_0xb6a9870 .scope module, "a0" "a" 2 45, 3 3, S_0xb67def0;
 .timescale 0 0;
L_0xb6aad00 .functor BUFZ 1, L_0xb6ab1f0, C4<0>, C4<0>, C4<0>;
L_0xb6aad60 .functor BUFZ 1, C4<z>, C4<0>, C4<0>, C4<0>;
L_0xb6aae50 .functor BUFZ 1, v0xb6aa050_0, C4<0>, C4<0>, C4<0>;
v0xb6a9950_0 .net "clk", 0 0, v0xb6aa050_0; 1 drivers
v0xb6a99d0_0 .alias "in", 0 0, v0xb6a9d10_0;
v0xb6a9a50_0 .alias "out", 0 0, v0xb6a9de0_0;
v0xb6a9ad0_0 .net "s_clk", 0 0, L_0xb6aae50; 1 drivers
v0xb6a9b50_0 .net "s_in", 0 0, L_0xb6aad00; 1 drivers
v0xb6a9bd0_0 .net "s_out", 0 0, L_0xb6aad60; 1 drivers
S_0xb6a7fd0 .scope module, "stim_a0" "stim_expect_mem_template" 2 55, 4 3, S_0xb67def0;
 .timescale 0 0;
P_0xb6a80b8 .param/l "ADDR_WIDTH" 4 12, +C4<010000>;
P_0xb6a80e0 .param/l "MEM_DEPTH" 4 18, C4<000000000000000001111111111111111>;
P_0xb6a8108 .param/l "MEM_WIDTH" 4 11, +C4<01>;
P_0xb6a8130 .param/str "VECTOR_FILE" 4 16, "stim";
P_0xb6a8158 .param/l "VECTOR_ID" 4 13, +C4<0>;
P_0xb6a8180 .param/str "VECTOR_NAME" 4 15, "stim";
P_0xb6a81a8 .param/l "VECTOR_RADIX" 4 17, +C4<0>;
P_0xb6a81d0 .param/l "VECTOR_VERSION" 4 14, +C4<0>;
L_0xb6aaf40 .functor OR 1, v0xb6a8f00_0, v0xb6a8e80_0, C4<0>, C4<0>;
L_0xb6ab000 .functor NOT 1, v0xb6aa650_0, C4<0>, C4<0>, C4<0>;
L_0xb6ab120 .functor OR 1, L_0xb6ab000, L_0xb6aaf40, C4<0>, C4<0>;
L_0xb6ab360 .functor XNOR 1, v0xb6a9090_0, C4<0>, C4<0>, C4<0>;
L_0xb6ab460 .functor AND 1, L_0xb6ab360, v0xb6a8f00_0, C4<1>, C4<1>;
L_0xb6ab5a0 .functor XNOR 1, v0xb6a9090_0, C4<0>, C4<0>, C4<0>;
L_0xb6ab690 .functor AND 1, L_0xb6ab5a0, v0xb6a8e80_0, C4<1>, C4<1>;
L_0xb6ab7d0 .functor XOR 1, v0xb6a9090_0, C4<0>, C4<0>, C4<0>;
L_0xb6ab910 .functor AND 1, v0xb6a8e80_0, L_0xb6ab7d0, C4<1>, C4<1>;
L_0xb6aba00 .functor XOR 1, v0xb6a9090_0, C4<0>, C4<0>, C4<0>;
L_0xb6abb10 .functor AND 1, v0xb6a8f00_0, L_0xb6aba00, C4<1>, C4<1>;
L_0xb6abb70 .functor NOT 1, L_0xb6aaf40, C4<0>, C4<0>, C4<0>;
L_0xb6abab0 .functor AND 1, v0xb6aa650_0, L_0xb6abb70, C4<1>, C4<1>;
v0xb6a84c0_0 .net *"_s10", 0 0, C4<0>; 1 drivers
v0xb6a8580_0 .net *"_s12", 0 0, L_0xb6ab360; 1 drivers
v0xb6a8620_0 .net *"_s16", 0 0, C4<0>; 1 drivers
v0xb6a86c0_0 .net *"_s18", 0 0, L_0xb6ab5a0; 1 drivers
v0xb6a8740_0 .net *"_s2", 0 0, L_0xb6ab000; 1 drivers
v0xb6a87e0_0 .net *"_s22", 0 0, C4<0>; 1 drivers
v0xb6a88c0_0 .net *"_s24", 0 0, L_0xb6ab7d0; 1 drivers
v0xb6a8960_0 .net *"_s28", 0 0, C4<0>; 1 drivers
v0xb6a8a50_0 .net *"_s30", 0 0, L_0xb6aba00; 1 drivers
v0xb6a8af0_0 .net *"_s34", 0 0, L_0xb6abb70; 1 drivers
v0xb6a8bf0_0 .net *"_s6", 0 0, C4<0>; 1 drivers
v0xb6a8c90_0 .alias "clock", 0 0, v0xb6a9950_0;
v0xb6a8d80_0 .alias "id_err", 0 0, v0xb6aa550_0;
v0xb6a8e00_0 .var/i "mem_addr", 31 0;
v0xb6a8f00_0 .var "mem_out_is_id", 0 0;
v0xb6a8f80_0 .net "mem_out_is_id_or_version", 0 0, L_0xb6aaf40; 1 drivers
v0xb6a8e80_0 .var "mem_out_is_version", 0 0;
v0xb6a9090_0 .var "memory_out", 0 0;
v0xb6a9000_0 .net "mux_select", 0 0, L_0xb6ab120; 1 drivers
v0xb6a91b0_0 .var "rd_addr", 15 0;
v0xb6a9130_0 .alias "rd_en", 0 0, v0xb6aa6d0_0;
v0xb6a9310_0 .net "reset_", 0 0, v0xb6aa860_0; 1 drivers
v0xb6a9260 .array "stim_expect_memory", 65535 0, 0 0;
v0xb6a9450_0 .var "stim_expect_memory_loaded", 0 0;
v0xb6a9390_0 .alias "valid", 0 0, v0xb6aa3e0_0;
v0xb6a95a0_0 .net "vector_id_match", 0 0, L_0xb6ab460; 1 drivers
v0xb6a94d0_0 .alias "vector_out", 0 0, v0xb6a9d10_0;
v0xb6a9700_0 .net "vector_version_match", 0 0, L_0xb6ab690; 1 drivers
v0xb6a9620_0 .alias "version_err", 0 0, v0xb6aa980_0;
E_0xb6a82a0 .event edge, v0xb6a9450_0;
L_0xb6ab1f0 .functor MUXZ 1, v0xb6a9090_0, C4<0>, L_0xb6ab120, C4<>;
S_0xb672070 .scope module, "exp_a0" "stim_expect_mem_template" 2 69, 4 3, S_0xb67def0;
 .timescale 0 0;
P_0xb67ac48 .param/l "ADDR_WIDTH" 4 12, +C4<010000>;
P_0xb67ac70 .param/l "MEM_DEPTH" 4 18, C4<000000000000000001111111111111111>;
P_0xb67ac98 .param/l "MEM_WIDTH" 4 11, +C4<01>;
P_0xb67acc0 .param/str "VECTOR_FILE" 4 16, "exp";
P_0xb67ace8 .param/l "VECTOR_ID" 4 13, +C4<0>;
P_0xb67ad10 .param/str "VECTOR_NAME" 4 15, "exp";
P_0xb67ad38 .param/l "VECTOR_RADIX" 4 17, +C4<0>;
P_0xb67ad60 .param/l "VECTOR_VERSION" 4 14, +C4<0>;
L_0xb6abc90 .functor OR 1, v0xb6a7610_0, v0xb6a7590_0, C4<0>, C4<0>;
L_0xb6abcf0 .functor NOT 1, v0xb6aa650_0, C4<0>, C4<0>, C4<0>;
L_0xb6abd50 .functor OR 1, L_0xb6abcf0, L_0xb6abc90, C4<0>, C4<0>;
L_0xb6ac0a0 .functor XNOR 1, v0xb6a77a0_0, C4<0>, C4<0>, C4<0>;
L_0xb6ac1a0 .functor AND 1, L_0xb6ac0a0, v0xb6a7610_0, C4<1>, C4<1>;
L_0xb6ac2a0 .functor XNOR 1, v0xb6a77a0_0, C4<0>, C4<0>, C4<0>;
L_0xb6ac390 .functor AND 1, L_0xb6ac2a0, v0xb6a7590_0, C4<1>, C4<1>;
L_0xb6ac530 .functor XOR 1, v0xb6a77a0_0, C4<0>, C4<0>, C4<0>;
L_0xb6ac620 .functor AND 1, v0xb6a7590_0, L_0xb6ac530, C4<1>, C4<1>;
L_0xb6ac710 .functor XOR 1, v0xb6a77a0_0, C4<0>, C4<0>, C4<0>;
L_0xb6ac820 .functor AND 1, v0xb6a7610_0, L_0xb6ac710, C4<1>, C4<1>;
L_0xb6ac910 .functor NOT 1, L_0xb6abc90, C4<0>, C4<0>, C4<0>;
L_0xb6ac7c0 .functor AND 1, v0xb6aa650_0, L_0xb6ac910, C4<1>, C4<1>;
v0xb687cd0_0 .net *"_s10", 0 0, C4<0>; 1 drivers
v0xb6a6c60_0 .net *"_s12", 0 0, L_0xb6ac0a0; 1 drivers
v0xb6a6d00_0 .net *"_s16", 0 0, C4<0>; 1 drivers
v0xb6a6da0_0 .net *"_s18", 0 0, L_0xb6ac2a0; 1 drivers
v0xb6a6e50_0 .net *"_s2", 0 0, L_0xb6abcf0; 1 drivers
v0xb6a6ef0_0 .net *"_s22", 0 0, C4<0>; 1 drivers
v0xb6a6fd0_0 .net *"_s24", 0 0, L_0xb6ac530; 1 drivers
v0xb6a7070_0 .net *"_s28", 0 0, C4<0>; 1 drivers
v0xb6a7160_0 .net *"_s30", 0 0, L_0xb6ac710; 1 drivers
v0xb6a7200_0 .net *"_s34", 0 0, L_0xb6ac910; 1 drivers
v0xb6a7300_0 .net *"_s6", 0 0, C4<0>; 1 drivers
v0xb6a73a0_0 .alias "clock", 0 0, v0xb6a9950_0;
v0xb6a7490_0 .alias "id_err", 0 0, v0xb6aa550_0;
v0xb6a7510_0 .var/i "mem_addr", 31 0;
v0xb6a7610_0 .var "mem_out_is_id", 0 0;
v0xb6a7690_0 .net "mem_out_is_id_or_version", 0 0, L_0xb6abc90; 1 drivers
v0xb6a7590_0 .var "mem_out_is_version", 0 0;
v0xb6a77a0_0 .var "memory_out", 0 0;
v0xb6a78c0_0 .net "mux_select", 0 0, L_0xb6abd50; 1 drivers
v0xb6a7940_0 .var "rd_addr", 15 0;
v0xb6a7820_0 .alias "rd_en", 0 0, v0xb6aa6d0_0;
v0xb6a7a70_0 .alias "reset_", 0 0, v0xb6a9310_0;
v0xb6a79c0 .array "stim_expect_memory", 65535 0, 0 0;
v0xb6a7bb0_0 .var "stim_expect_memory_loaded", 0 0;
v0xb6a7af0_0 .alias "valid", 0 0, v0xb6aa3e0_0;
v0xb6a7d00_0 .net "vector_id_match", 0 0, L_0xb6ac1a0; 1 drivers
v0xb6a7c30_0 .alias "vector_out", 0 0, v0xb6a9e60_0;
v0xb6a7e60_0 .net "vector_version_match", 0 0, L_0xb6ac390; 1 drivers
v0xb6a7d80_0 .alias "version_err", 0 0, v0xb6aa980_0;
E_0xb67bab0 .event edge, v0xb6a7bb0_0;
E_0xb67bb20/0 .event negedge, v0xb6a7a70_0;
E_0xb67bb20/1 .event posedge, v0xb6a73a0_0;
E_0xb67bb20 .event/or E_0xb67bb20/0, E_0xb67bb20/1;
L_0xb6abe90 .functor MUXZ 1, v0xb6a77a0_0, C4<0>, L_0xb6abd50, C4<>;
    .scope S_0xb6a7fd0;
T_0 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6a9310_0, 1;
    %inv 8, 1;
    %jmp/0xz  T_0.0, 8;
    %ix/load 0, 16, 0;
    %assign/v0 v0xb6a91b0_0, 0, 0;
    %jmp T_0.1;
T_0.0 ;
    %load/v 8, v0xb6a9130_0, 1;
    %jmp/0xz  T_0.2, 8;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6a91b0_0, 16;
    %ix/load 0, 16, 0;
    %assign/v0 v0xb6a91b0_0, 0, 8;
    %load/v 8, v0xb6a91b0_0, 16;
    %cmpi/u 8, 0, 16;
    %mov 8, 4, 1;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a8f00_0, 0, 8;
    %load/v 8, v0xb6a91b0_0, 16;
    %cmpi/u 8, 1, 16;
    %mov 8, 4, 1;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a8e80_0, 0, 8;
T_0.2 ;
T_0.1 ;
    %jmp T_0;
    .thread T_0;
    .scope S_0xb6a7fd0;
T_1 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6a9130_0, 1;
    %jmp/0xz  T_1.0, 8;
    %ix/getv 3, v0xb6a91b0_0;
    %load/av 8, v0xb6a9260, 1;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a9090_0, 0, 8;
T_1.0 ;
    %jmp T_1;
    .thread T_1;
    .scope S_0xb6a7fd0;
T_2 ;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a9450_0, 0, 0;
    %vpi_call 4 66 "$display", "VECTOR_FILE= %s", P_0xb6a8130;
    %vpi_call 4 68 "$readmemb", P_0xb6a8130, v0xb6a9260;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a9450_0, 0, 1;
    %end;
    .thread T_2;
    .scope S_0xb6a7fd0;
T_3 ;
    %wait E_0xb6a82a0;
    %vpi_func 4 79 "$test$plusargs", 8, 32, "show_stim_expect_memory_init_state";
    %cmpi/u 8, 0, 32;
    %inv 4, 1;
    %jmp/0xz  T_3.0, 4;
    %vpi_call 4 80 "$display", "Initial state of vector file %s ", P_0xb6a8130;
    %set/v v0xb6a8e00_0, 0, 32;
T_3.2 ;
    %load/v 8, v0xb6a8e00_0, 32;
    %mov 40, 0, 1;
   %cmpi/u 8, 65535, 33;
    %jmp/0xz T_3.3, 5;
    %vpi_call 4 83 "$display", "mem[%d] = %x", v0xb6a8e00_0, &A<v0xb6a9260, v0xb6a8e00_0 >;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6a8e00_0, 32;
    %set/v v0xb6a8e00_0, 8, 32;
    %jmp T_3.2;
T_3.3 ;
T_3.0 ;
    %end;
    .thread T_3;
    .scope S_0xb672070;
T_4 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6a7a70_0, 1;
    %inv 8, 1;
    %jmp/0xz  T_4.0, 8;
    %ix/load 0, 16, 0;
    %assign/v0 v0xb6a7940_0, 0, 0;
    %jmp T_4.1;
T_4.0 ;
    %load/v 8, v0xb6a7820_0, 1;
    %jmp/0xz  T_4.2, 8;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6a7940_0, 16;
    %ix/load 0, 16, 0;
    %assign/v0 v0xb6a7940_0, 0, 8;
    %load/v 8, v0xb6a7940_0, 16;
    %cmpi/u 8, 0, 16;
    %mov 8, 4, 1;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a7610_0, 0, 8;
    %load/v 8, v0xb6a7940_0, 16;
    %cmpi/u 8, 1, 16;
    %mov 8, 4, 1;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a7590_0, 0, 8;
T_4.2 ;
T_4.1 ;
    %jmp T_4;
    .thread T_4;
    .scope S_0xb672070;
T_5 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6a7820_0, 1;
    %jmp/0xz  T_5.0, 8;
    %ix/getv 3, v0xb6a7940_0;
    %load/av 8, v0xb6a79c0, 1;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a77a0_0, 0, 8;
T_5.0 ;
    %jmp T_5;
    .thread T_5;
    .scope S_0xb672070;
T_6 ;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a7bb0_0, 0, 0;
    %vpi_call 4 66 "$display", "VECTOR_FILE= %s", P_0xb67acc0;
    %vpi_call 4 68 "$readmemb", P_0xb67acc0, v0xb6a79c0;
    %ix/load 0, 1, 0;
    %assign/v0 v0xb6a7bb0_0, 0, 1;
    %end;
    .thread T_6;
    .scope S_0xb672070;
T_7 ;
    %wait E_0xb67bab0;
    %vpi_func 4 79 "$test$plusargs", 8, 32, "show_stim_expect_memory_init_state";
    %cmpi/u 8, 0, 32;
    %inv 4, 1;
    %jmp/0xz  T_7.0, 4;
    %vpi_call 4 80 "$display", "Initial state of vector file %s ", P_0xb67acc0;
    %set/v v0xb6a7510_0, 0, 32;
T_7.2 ;
    %load/v 8, v0xb6a7510_0, 32;
    %mov 40, 0, 1;
   %cmpi/u 8, 65535, 33;
    %jmp/0xz T_7.3, 5;
    %vpi_call 4 83 "$display", "mem[%d] = %x", v0xb6a7510_0, &A<v0xb6a79c0, v0xb6a7510_0 >;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6a7510_0, 32;
    %set/v v0xb6a7510_0, 8, 32;
    %jmp T_7.2;
T_7.3 ;
T_7.0 ;
    %end;
    .thread T_7;
    .scope S_0xb67def0;
T_8 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6aa860_0, 1;
    %inv 8, 1;
    %jmp/0xz  T_8.0, 8;
    %ix/load 0, 32, 0;
    %assign/v0 v0xb6aa0d0_0, 0, 0;
    %jmp T_8.1;
T_8.0 ;
    %load/v 8, v0xb6aa0d0_0, 32;
    %mov 40, 39, 1;
   %addi 8, 1, 33;
    %ix/load 0, 32, 0;
    %assign/v0 v0xb6aa0d0_0, 0, 8;
T_8.1 ;
    %jmp T_8;
    .thread T_8;
    .scope S_0xb67def0;
T_9 ;
    %set/v v0xb6aa050_0, 0, 1;
    %set/v v0xb6aa650_0, 1, 1;
    %set/v v0xb6aa860_0, 1, 1;
    %delay 10, 0;
    %set/v v0xb6aa860_0, 0, 1;
    %delay 20, 0;
    %set/v v0xb6aa860_0, 1, 1;
    %vpi_func 2 94 "$fopen", 8, 32, "vectors.txt";
    %set/v v0xb6aa4d0_0, 8, 32;
    %load/v 8, v0xb6aa4d0_0, 32;
    %cmpi/u 8, 0, 32;
    %jmp/0xz  T_9.0, 4;
    %vpi_call 2 96 "$display", "Error opening vectors.txt file";
    %vpi_call 2 97 "$finish";
T_9.0 ;
    %vpi_call 2 99 "$display", v0xb6aa4d0_0, "Dut outputs vs expected vectors:\012";
    %vpi_call 2 100 "$dumpfile", "wavesDefaultOutputFile_dump";
    %vpi_call 2 101 "$dumpvars", 1'sb0, S_0xb67def0;
    %vpi_call 2 102 "$dumpon";
    %set/v v0xb6aa1a0_0, 0, 16;
    %set/v v0xb6aa240_0, 0, 16;
    %set/v v0xb6aa340_0, 0, 16;
    %end;
    .thread T_9;
    .scope S_0xb67def0;
T_10 ;
    %delay 20, 0;
    %load/v 8, v0xb6aa050_0, 1;
    %inv 8, 1;
    %set/v v0xb6aa050_0, 8, 1;
    %jmp T_10;
    .thread T_10;
    .scope S_0xb67def0;
T_11 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6aa860_0, 1;
    %inv 8, 1;
    %jmp/0xz  T_11.0, 8;
    %set/v v0xb6aa240_0, 0, 16;
    %jmp T_11.1;
T_11.0 ;
    %load/v 8, v0xb6a9f90_0, 1;
    %jmp/0xz  T_11.2, 8;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6aa340_0, 16;
    %set/v v0xb6aa340_0, 8, 16;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6aa240_0, 16;
    %set/v v0xb6aa240_0, 8, 16;
    %vpi_call 2 118 "$display", "mismatch detected: dut %s shows value %b; evepect vector %s shows value %b\012 ", "a0", v0xb6a9de0_0, "exp_a0", v0xb6a9e60_0;
T_11.2 ;
    %load/v 8, v0xb6aa240_0, 16;
    %cmp/u 0, 8, 16;
    %jmp/0xz  T_11.4, 5;
    %vpi_call 2 121 "$display", "Maximum number or errors allowed for vector %s has been reached", "exp_a0";
T_11.4 ;
T_11.1 ;
    %jmp T_11;
    .thread T_11;
    .scope S_0xb67def0;
T_12 ;
    %wait E_0xb67bb20;
    %load/v 8, v0xb6aa860_0, 1;
    %inv 8, 1;
    %jmp/0xz  T_12.0, 8;
    %set/v v0xb6aa1a0_0, 0, 16;
    %jmp T_12.1;
T_12.0 ;
    %load/v 8, v0xb6a9f10_0, 1;
    %jmp/0xz  T_12.2, 8;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6aa340_0, 16;
    %set/v v0xb6aa340_0, 8, 16;
    %ix/load 0, 1, 0;
    %load/vp0 8, v0xb6aa1a0_0, 16;
    %set/v v0xb6aa1a0_0, 8, 16;
    %vpi_call 2 134 "$display", "match detected: dut %s shows value %b; evepect vector %s shows value %b\012 ", "a0", v0xb6a9de0_0, "exp_a0", v0xb6a9e60_0;
T_12.2 ;
T_12.1 ;
    %jmp T_12;
    .thread T_12;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "tb_testbench.v";
    "a.v";
    "stim_expect_mem_template.v";
